### ISSUE TITLE
The page isn’t redirecting properly

### ISSUE THREAD

**Issue by crssi** [snyderp/web-api-manager#27]
I have a page that gets this error even if I clear all "blocked standards":

```
The page isn’t redirecting properly
Firefox has detected that the server is redirecting the request for this address in a way that will never complete.
This problem can sometimes be caused by disabling or refusing to accept cookies.
```

From log:
` [HTTP/1.1 302 Moved Temporarily]`

The only way making it work is to disable WAM,

If would like not to reveal URL here.

**Comment by snyderp** [comment]
Hmm, two questions that'd help me figure out what could be going on:

1. Are you sure you're using the most recent release?
2. If you'd prefer to not give out the URL, can you create a test case that replicates the issue, or point to some other site that triggers the issue?  Or, if you'd prefer to not reveal the URL here, but would be ok sharing it in a less public forum, we can arrange another method to communicate it

**Comment by snyderp** [comment]
I'm going to close this for now.  If there is a way I can recreate, or that you can let me know in the future what site is causing the problems though, please let me know and I'll get it sorted out then.  Thanks!

**Comment by crssi** [comment]
I am out of the country until the end of the week with a very limited internet access. Will respond properly when back.
Cheers

**Comment by crssi** [comment]
@snyderp Can I use your uic.edu email for sending you more details?

**Comment by snyderp** [comment]
Yep, that'd be great.   If further privacy would be good, feel free to use my PGP key

https://www.cs.uic.edu/~psnyder/static/misc/psnyde2uic.edu.pub.asc

**Comment by crssi** [comment]
Have you got the email?

**Comment by snyderp** [comment]
Yep!  It’s in my queue, on it :)

**Comment by crssi** [comment]
Any ETA for this one?

Cheers

**Comment by snyderp** [comment]
I've narrowed the issue down to the issue with the site using http links, but being automatically redirected to https because of the HSTS header.  This is a bug in the site for sure, and would be best if they could fix things (if you have a connection), but in the meantime I do not have a simple solution in Firefox.

Could you try on Chrome and let me know if you see the issue there.  If not, I'll open another bug with the Firefox folks, to give extensions some hook to manage this.

Thanks for your patience and help!

**Comment by crssi** [comment]
On Chrome it works... but, then I have start to dig into FF, since there is no problem also with clean profile.

When WAM is installed (regardles of the settings) there is a breakage, if any of the following is true:
1. Pref `network.http.referer.XOriginPolicy` not equal to `0`
2. uBlock is installed

But if there is no WAM, then `network.http.referer.XOriginPolicy` and uBO doesn't matter, it works.

Is WAM injecting any script to a opening page?
If so, does it inject a script even if none of the settings are checked?

**Comment by snyderp** [comment]
Thank you for looking more into this.  This is all extremely helpful!

To answer your questions:

1. WAM injects the [blocking script](https://github.com/snyderp/web-api-manager/blob/master/add-on/lib/proxyblock.js) into pages.  This script returns quick if there are no standards to be blocked, but otherwise performs its instrumentation, and then removes itself from the DOM.  

So, this brings up an interesting bug / fix that I didn't realize, if there are no standards to block in the frame, then the script early exits, but *doesn't* remove itself from the page.  I don't think this should have an effect on the issue you're seeing, but its definitely sub-optional, and should be fixed.

And since I don't yet have an explanation for the issue you're seeing,  maybe this will fix it ¯\_(ツ)_/¯

I'll open an issue and get that sorted in the next release.

In the meantime, does anything I said above give you a guess about what might be going on?

(either way, i'll reopen the issue, since its clear somethings not right)

**Comment by crssi** [comment]
Yes it does, this page is foobar anyway. ;)
But, even so, it shines a light to even better implementation. ;)

I wouldn't bother with this page anyway.

**Comment by snyderp** [comment]
Ok then, closing.  But if you notice the issue on other pages, please give me a heads up.  It might be a bug in Mozilla's WebExtension system.  Thanks!

**Comment by crssi** [comment]
Is it possible to implement when a domain is defined without any blocking rules, that WAM would not interfere at all.
It is not a good, but at least it is a solution to temporarily workaround a few pages to make them work until this is sorted out on higher level.

Cheers

**Comment by snyderp** [comment]
@crssi yep!  Thats definitely possible!  I'll have that change in the next release.  Great, easy improvement.  Thanks!

**Comment by crssi** [comment]
Another one fits to this category:
Microsoft Exchange 2013 Control Panel web... the properties pane loads indefinitely.
Unfortunately, for this one I cannot provide URL. :(

Do you have any ETA for next release?
It would help to have WAM always running in order to find some other inconveniences.

**Comment by snyderp** [comment]
I am going to try hell or high water to get the next release out tonight.  I'm try to get a lot more tests in place, since a lot of things have shifted since the last release.  But, I'm on it.  24 hours or yr $$$ back :P

**Comment by crssi** [comment]
Don't kill yourself. I can wait longer if needed. ;)

Cheers

**Comment by snyderp** [comment]
Ok, its out, phew!  If you could let me know if it fixes (or at least lessens) your issues, I'd greatly appreciate it :)

**Comment by crssi** [comment]
Whitelisting now resolves #55 and [https://github.com/snyderp/web-api-manager/issues/27#issuecomment-351050202](https://github.com/snyderp/web-api-manager/issues/27#issuecomment-351050202).
Good job, thank you

It does not resolves [https://github.com/snyderp/web-api-manager/issues/27#issue-269375447](https://github.com/snyderp/web-api-manager/issues/27#issue-269375447) (initial post), but here the page itself is a bit foobar.

Cheers

**Comment by snyderp** [comment]
Eggcellent :)
